scripts/json_overview_image_info.py

```diff
@@ -18,27 +18,41 @@
 
 output = {}
 
+
+def get_initial_output(image_info):
+    # preserve existing profiles.json
+    if output_path.is_file():
+        profiles = json.loads(output_path.read_text())
+        if profiles["version_code"] == image_info["version_code"]:
+            return profiles
+    return image_info
+
+
 for json_file in work_dir.glob("*.json"):
     image_info = json.loads(json_file.read_text())
+
     if not output:
-        output.update(image_info)
+        output = get_initial_output(image_info)
+
+    # get first and only profile in json file
+    device_id, profile = next(iter(image_info["profiles"].items()))
+    if device_id not in output["profiles"]:
+        output["profiles"][device_id] = profile
     else:
-        # get first (and only) profile in json file
-        device_id = next(iter(image_info["profiles"].keys()))
-        if device_id not in output["profiles"]:
-            output["profiles"].update(image_info["profiles"])
-        else:
-            output["profiles"][device_id]["images"].append(
-                image_info["profiles"][device_id]["images"][0]
-            )
+        output["profiles"][device_id]["images"].extend(profile["images"])
+
+# make image lists unique by name, keep last/latest
+for device_id, profile in output["profiles"].items():
```

Comment:
When we build the same images the same time, then we need to remove duplicate entries.

Reply:
I think a better approach here is to replace the profile entirely rather than adding images.